### ISSUE TITLE
Xcode 15.3 & 15.4 compilation issue with v10.29.0 using Carthage

### ISSUE THREAD

**Issue by henry42** [firebase/firebase-ios-sdk#13718]
### Description

I believe the issue is similar to the one described in https://github.com/firebase/firebase-ios-sdk/issues/12800. The binary seems to be built using Swift 5.9, while Xcode 15.3+ uses Swift 5.10.

I hope that the Firebase SDK without the Swift-suffixed binary should not be impacted by the Swift language version in Xcode.

At least, version 10.29.0 should broadly support the current Xcode version, especially since it is the last version that supports iOS 12.

Thank you so much for this project; I truly depend on it.

### Reproducing the issue

binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAnalyticsBinary.json" == 10.29.0
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseCrashlyticsBinary.json" == 10.29.0
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseInAppMessagingBinary.json" == 10.29.0
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseMessagingBinary.json" == 10.29.0
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseRemoteConfigBinary.json" == 10.29.0

### Firebase SDK Version

10.29.0

### Xcode Version

15.4

### Installation Method

Carthage

### Firebase Product(s)

Remote Config

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Failed to build module 'FirebaseRemoteConfig'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.9... while this compiler is 'Apple Swift version 5.10.... Please select a toolchain which matches the SDK.
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
This may be related to https://github.com/firebase/firebase-ios-sdk/issues/12877. Please make sure you delete all old Firebase libraries and follow all the Carthage instructions.

Since we're not able to reproduce with our testing, we'll need a full reproducible example.

**Comment by henry42** [comment]
Yes, I deleted all old Firebase libraries and clean the building folder. And I am sure that it is easy to reproduce, just use Xcode 15.3 or 15.4 and Carthage with Firebase 10.29.0

**Comment by paulb777** [comment]
Closed for staleness. We haven't been able to reproduce and Apple will soon stop supporting Xcode 15.
